models/Book.js

```diff
@@ -0,0 +1,103 @@
+const mongoose = require('mongoose')
+
+const BookSchema = new mongoose.Schema(
+    {
+        //options which should be considered when sharing a book!
+
+        userName: {
+            type: String,
+            required: true,
+        },
```

Comment:
We don't actually need to include userName in the model. Rather for the queries can you populate the userName based on the userId. Check documentation [here](https://mongoosejs.com/docs/populate.html#field-selection)